skimage/restoration/inpaint.py

```diff
@@ -1,79 +1,200 @@
+    of pixels to be inpainted). Each row in A will have a number of non-zero
+    values equal to the number of non-zero values in the biharmonic kernel,
+    ``neigh_coef_full``. In practice, biharmonic kernels with reduced extent
+    are used at the image borders. This matrix, ``A`` is the same for all
+    image channels (since the same inpainting mask is currently used for all
+    channels).
+
+    ``u`` is a dense matrix of shape ``(n_mask, n_channels)`` and represents
+    the vector of unknown values for each channel.
+
+    ``b`` is a dense matrix of shape ``(n_mask, n_channels)`` and represents
+    the desired output of convolving the solution with the biharmonic kernel.
+    At mask locations where there is no overlap with known values, ``b`` will
+    have a value of 0. This enforces the biharmonic smoothness constraint in
+    the interior of inpainting regions. For regions near the boundary that
+    overlap with known values, the entries in ``b`` enforce boundary conditions
+    designed to avoid discontinuity with the known values.
+    """
+
+    n_channels = out.shape[-1]
+    radius = neigh_coef_full.shape[0] // 2
+
+    edge_mask = np.ones(mask.shape, dtype=bool)
+    edge_mask[(slice(radius, -radius),) * mask.ndim] = 0
+    boundary_mask = edge_mask * mask
+    center_mask = ~edge_mask * mask
+
+    boundary_pts = np.where(boundary_mask)
+    boundary_i = np.flatnonzero(boundary_mask)
+    center_i = np.flatnonzero(center_mask)
+    mask_i = np.concatenate((boundary_i, center_i))
+
+    center_pts = np.where(center_mask)
+    mask_pts = tuple(
+        [np.concatenate((b, c)) for b, c in zip(boundary_pts, center_pts)]
+    )
+
+    # Use convolution to predetermine the number of non-zero entries in the
+    # sparse system matrix.
+    structure = neigh_coef_full != 0
+    tmp = ndi.convolve(mask, structure, output=np.uint8, mode='constant')
+    nnz_matrix = tmp[mask].sum()
+
+    # Need to estimate the number of zeros for the right hand side vector.
+    # The computation below will slightly overestimate the true number of zeros
+    # due to edge effects (the kernel itself gets shrunk in size near the
+    # edges, but that isn't accounted for here). We can trim any excess entries
+    # later.
+    n_mask = np.count_nonzero(mask)
+    n_struct = np.count_nonzero(structure)
+    nnz_rhs_vector_max = n_mask - np.count_nonzero(tmp == n_struct)
+
+    # pre-allocate arrays storing sparse matrix indices and values
+    row_idx_known = np.empty(nnz_rhs_vector_max, dtype=np.intp)
+    data_known = np.zeros((nnz_rhs_vector_max, n_channels), dtype=out.dtype)
+    row_idx_unknown = np.empty(nnz_matrix, dtype=np.intp)
+    col_idx_unknown = np.empty(nnz_matrix, dtype=np.intp)
+    data_unknown = np.empty(nnz_matrix, dtype=out.dtype)
+
+    # cache the various small, non-square Laplacians used near the boundary
+    coef_cache = {}
+
+    # Iterate over masked points near the boundary
+    mask_flat = mask.ravel()
```

Comment:
```suggestion
    mask_flat = mask.reshape(-1)
```
See the note section [here](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html)